content/tutorials/token-authentication.textile

```diff
@@ -207,11 +268,50 @@ blang[ruby].
 
   "See this step in Github":https://github.com/ably/tutorials/commit/5575d73
 
+blang[python].
+  By adding the following class and route to your urls, it is now ready to service clients wanting to authenticate with Ably.
+
+  ```[python]
+    import json
+
+    # Issue token requests to clients sending a request to the /auth endpoint
+    class auth:
+        def GET(self):
+            web.header('Content-Type', 'application/json')
+            return json.dumps(client.auth.create_token_request().to_dict())
+
+    # let's update urls
+    urls = (
+        '/', index,
+        '/auth', auth
+    )
+  ```
+
+  Try running your server again with @python server.py@ and visiting "http://localhost:8080/auth":http://localhost:8080/auth and you should see a JSON token request in the form:
+
+  ```[json]
+  {
+    "keyName": "I2ACJQ.Lv8AUQ",
+    "ttl": 3600000,
+    "timestamp": 1473894038255,
+    "capability": "{\"*\":[\"*\"]}",
+    "nonce": "f6799a8e7fa6f77b8e1dac55314789b1",
+    "mac": "35nifY9SRZ8KRDfKOPIS1qYWGP16r2lD59zJo9TH8pA="
+  }
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/7841d7e
+
 h2. Step 5 - Setting up a client to use token authentication
 
 Now that we have a web server running and an authentication endpoint, we are ready to setup the Ably Realtime client that uses token auth.
 
-First we need to set up a simple HTML page that will be served by the web server. Create the following HTML file named @index.html@ in the same folder as @server.js@:
+blang[nodejs].
+  First we need to set up a simple HTML page that will be served by the web server. Create the following HTML file named @index.html@ in the same folder as @server.js@:
+blang[ruby].
```

Comment:
I believe you need a single line of whitespace for `blang` blocks. Are you running this locally and have you confirmed it looks right when the page is rendered?  this is essential please